examples/asyncio/asyncio-server.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+"""
+asyncio-server.py
+~~~~~~~~~~~~~~~~~
+
+A fully-functional HTTP/2 server using asyncio. Requires Python 3.5+.
+"""
+import asyncio
+import json
+import ssl
+import collections
+from typing import List, Tuple
+
+from h2.connection import H2Connection
+from h2.events import DataReceived, RequestReceived, RemoteSettingsChanged
+
+
+class H2Protocol(asyncio.Protocol):
+    def __init__(self):
+        self.conn = H2Connection(client_side=False)
+        self.transport = None
+
+    def connection_made(self, transport: asyncio.Transport):
+        self.transport = transport
+        self.conn.initiate_connection()
+        self.transport.write(self.conn.data_to_send())
+
+    def data_received(self, data: bytes):
+        events = self.conn.receive_data(data)
+        if self.conn.data_to_send:
```

Comment:
Yeah, you're right, that's probably not a sensible check. ;) It's left behind from before hyper-h2 had a proper interface for sending data.

I've substantially rewritten the Twisted example today because I'm at PyCon UK and I want to have something interesting to show off. Feel free to diverge from the Twisted example if you want!